File: src/1750. Minimum Length of String After Deleting Similar Ends/1750.go
func minimumLength(s string) int {
	left, right := 0, len(s)-1
	for left < right && s[left] == s[right] {
		c := s[left]
		for left <= right && s[left] == c {
			left++
		}
		for right >= left && s[right] == c {
			right--
		}
	}
	return right - left + 1
}

File: src/1750. Minimum Length of String After Deleting Similar Ends/1750.py
class Solution:
    def minimumLength(self, s: str) -> int:
        left, right = 0, len(s) - 1
        while left < right and s[left] == s[right]:
            c = s[left]
            while left <=right and s[left] == c:
                left += 1
            while left <= right and s[right] == c:
                right -= 1
        return right - left + 1
    
if __name__ == "__main__":
    